wordpress-mobile/WordPress-Android
# [Dependency Updates] Update `zendeskVersion` to 5.1.1

Parent #17569

This PR updates `zendeskVersion` to [5.1.1](https://developer.zendesk.com/documentation/classic-web-widget-sdks/support-sdk/android/release_notes/#511), plus a small dependency configuration refactor (d9ccdf4693dd72c1a27fe9ebe578bf651dc855a5).

Also, as part of this update the below transitive dependencies were added:
- On the `WordPress` module:
    - `com.squareup.retrofit2:retrofit` (89789974754f35a1aa80b84971dfc0569dc8cb0a)
    - `com.google.android.flexbox:flexbox` (8a7dc7e034e454401be6de62ecc392878b5f204b)

Additionally, the below exclusions got removed from the `com.zendesk:support` dependency:
- `com.android.support:support-annotations` (4a099b859a4bf0ed0fed342230bf42af5eda4409)
- `com.google.dagger` (37cacdad6c71f245fc445c0dbf47d171a19fc791)

Finally, the below issues got resolved:
- Lint: b1fa0b0ed1acef9041cb86bc0f01f641bbaf1b60
- Unit Test: 57a705b1ef4a22de465d45e1001cb5f9457e4903

-----

PS: @ovitrif I added you as the main reviewer, randomly, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

## To test:

1. See the dependency tree diff result and verify correctness.
2. Thoroughly smoke test any Zendesk related functionality, on both, the WordPress and Jetpack apps, and see if they both work as expected.
3. In addition to the above smoke test, you can expand the below and follow the inner and more explicit test steps within:

<details>
    <summary>1. Zendesk Update [Tickets & Contact Support]</summary>

ℹ️ This test only works when testing with a normal, non-a8c user account.

- Go to `Me` screen -> `Help` -> `Tickets`.
- Verify that the `My tickets` Zendesk related screen opens and that it lists all of your tickets (or non if you don't have any yet).
- Go back to the `Help` settings screen and then click on `Contact Support`.
- Verify that the `Contact Support` Zendesk related opens and that you are prompt to `Write a message`.
- Type `Testing Zendesk 5.1.1 update, please ignore and thank you! 🙏 🌟 🍀` and then send your message.
- Go back to the `Help` settings screen and then click on `Tickets` again.
- Verify that your previously sent message is listed in there and that you can click on it. PS: You could also check your emails and verify that you got a `Thank you for contacting the WordPress.com support team! (#1234567)` email.
- Verify that clicking on it navigates you to the inner screen where your message is shown as `Delivered` along with a predefined automated `mobile support` response message.

</details>

<details>
    <summary>2. Retrofit Update [TenorGifClient.kt]</summary>

- Go to `Media` screen and add a new `GIF` via the `Choose from Tenor` option (top right).
- Search to find GIFs to add to your media library. For example, type `something` and choose 3 GIFs from the resulting grid list, then click on the check option (top right) and wait for the GIFs to be uploaded.
- Verify that the GIFs you uploaded are being previewed as moving from the grid list of `ALL` media.
- Verify that clicking on a specific GIT you are navigated to its `File Details` and that it is being previewed as moving there too. 

</details>

<details>
    <summary>2. Flexbox Layout Update [BloggingPromptsOnboardingDialogFragment.kt]</summary>

ℹ️ This test only works when testing with the Jetpack app.

- While on the `HOME` sub-tab of the main `My Site` tab.
- Verify that the blogging prompts card are showing, displayed and as functional as expected. For example, try answering the prompt.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour on any or all Zendesk related app functionalities, like contacting support and/or viewing your support related tickets.

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

5. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
